dattri/task.py

```diff
@@ -123,20 +167,14 @@ def _load_checkpoints(self, ckpt_idx: int) -> None:
             self.current_checkpoint_idx is None
             or self.current_checkpoint_idx != ckpt_idx
         ):
-            if isinstance(self.checkpoints[ckpt_idx], (str, PosixPath)):
-                self.model.load_state_dict(
-                    torch.load(
-                        self.checkpoints[ckpt_idx],
-                        map_location=next(self.model.parameters()).device,
-                    ),
-                )
-            else:
-                self.model.load_state_dict(self.checkpoints[ckpt_idx])
+            self.model = self.checkpoints_load_func(
+                self.model,
+                self.checkpoints[ckpt_idx],
+            )
             self.current_checkpoint_idx = ckpt_idx
             self.named_parameters = {
                 k: p for k, p in self.model.named_parameters() if p.requires_grad
             }
-        self.model.eval()
```

Comment:
what is the original purpose of this line?

Reply:
When model is loaded, it will be changed to eval mode. Now it's in `_default_checkpoint_load_func`.

Reply:
got it